src/dvc_data/fs.py

```diff
@@ -151,9 +151,9 @@ def ls(self, path: "AnyFSPath", detail: bool = True, **kwargs: Any):
         root_key = self._get_key(path)
         try:
             info = self.index.info(root_key)
-            if info["type"] != "directory":
-                raise NotADirectoryError(path)
-
+            if info["type"] == "file":
+                info["name"] = path = self.join(*root_key)
+                return [info] if detail else [path]
```

Comment:
Only noteworthy change in this PR. `fs.ls()` should not raise error in this case, and should return information. 

This will also need to be handled properly in dvc side.